pandas/plotting/_matplotlib/converter.py

```diff
@@ -67,7 +67,13 @@ def register(explicit=True):
         converter = cls()
         if type_ in units.registry:
             previous = units.registry[type_]
-            _mpl_units[type_] = previous
+            # Exclude our converters from caching to make this idempotent.
+            if type(previous) not in {
```

Comment:
the preferred way to do this check would be `if not isinstance(previous, ...)`

Reply:
... though now that i look below, it looks like someone else used this pattern before, so never mind